ChildProject/pipelines/samplers.py

```diff
@@ -256,26 +268,65 @@ class HighVolubilitySampler(Sampler):
     def __init__(self,
         project: ChildProject.projects.ChildProject,
         annotation_set: str,
-        target_speaker_type: list,
-        windows_length: float,
+        windows_length: int,
         windows_count: int):
 
         super().__init__(project)
         self.annotation_set = annotation_set
-        self.target_speaker_type = target_speaker_type
         self.windows_length = windows_length
         self.windows_count = windows_count
 
+    def _segment_scores(self, recording):
+        segments = self.retrieve_segments(recording['recording_filename'])
+
+        if segments is None:
+            return pd.DataFrame(columns = ['segment_onset', 'segment_offset', 'recording_filename'])
+
+        # NOTE: The timestamps were simply rounded to 5 minutes in the original 
+        # code via a complicated string replacement , which imo was incorrect, but 
+        # there are edge cases that must be decided (even though they are quiet small)
+        segments['chunk'] = (segments['segment_offset']  // self.windows_length + 1).astype('int')
+
+        segment_onsets = segments.groupby('chunk')['segment_onset'].min()
+        segment_offsets = segments.groupby('chunk')['segment_offset'].max()
+
+        # this dataframe contains the segment onset and offsets for the chunks we calculated.
+        windows = pd.merge(segment_onsets, segment_offsets, left_index=True, right_index=True).reset_index()
+        windows['recording_filename'] = recording['recording_filename']
+
+        # NOTE: The original code explicitly chooses for these conversational turn 
+        # types, but we might wish to verify what they are. 
+        segments['is_CT'] = segments['lena_conv_turn_type'].isin(['TIFR', 'TIMR'])
```

Comment:
According to https://rdrr.io/github/HomeBankCode/DarcleITS/man/rlena.html:

> When the speaker changes from one segment to the other, this will (under certain circumstances) be counted as a strong conversational turn. Conversational turns can be of different types, too:
> 
> - `TIFI/TIMI` - Turn Initiation with Female / Male Adult
> - `TIFR/TIMR` - Turn Response with Female / Male Adult
> - `TIFE/TIME` - Turn End with Female / Male Adult
> - `NT` - Other Child with Key Child and Adult (Turns)

Reply:
Cool, then this seems correct for counting Conversational Turn Count (CTC) info.